packages/regl-worldview/src/commands/Command.js

```diff
@@ -127,35 +136,102 @@ export default class Command<T> extends React.Component<Props<T>> {
+        });
+      }
     }
     return memo;
   }, []);
 }
 
+// TODO: deprecating, remove before 1.x release
+function defaultGetObjectFromHitmapId(objectId: number, hitmapProps) {
+  return hitmapProps.find((hitmapProp) => {
+    if (hitmapProp.color) {
+      const hitmapPropId = getIdFromColor(hitmapProp.color.map((color) => color * 255));
+      if (hitmapPropId === objectId) {
+        return true;
+      }
+    }
+    return false;
+  });
+}
+
 // Factory function for creating simple regl components.
 // Sample usage: const Cubes = makeCommand('Cubes', rawCommand)
 // When you have children as the drawProps input, it's useful to simply call makeCommand
 // which creates a new regl component. It also handles basic hitmap interactions.
-export function makeCommand<T>(name: string, command: RawCommand<T>): React.StatelessFunctionalComponent<T> {
-  const cmd = (props: Props<T>) => {
-    const hitmapProps = props.getHitmapProps
-      ? props.getHitmapProps()
-      : getHitmapProps(props.getHitmapId, props.children);
-    return <Command {...props} reglCommand={command} drawProps={props.children} hitmapProps={hitmapProps} />;
+// use 'options' to control the default hitmap inputs
+export function makeCommand<T>(
+  name: string,
+  command: RawCommand<T>,
+  options: ?MakeCommandOptions = {}
+): React.StatelessFunctionalComponent<T> {
+  const cmd = ({
+    children,
+    enableInstanceHitmap,
+    enableHitmap,
+    getObjectFromHitmapId: getObjectFromHitmapIdAlt,
+    getHitmapProps: getHitmapPropsAlt,
+    getHitmapId,
+    ...rest
+  }: Props<T>) => {
+    let getObjectFromHitmapId = getObjectFromHitmapIdAlt || options.getObjectFromHitmapId;
+    let getHitmapProps = getHitmapPropsAlt || options.getHitmapProps;
+    let hitmapProps;
+
+    if ((getHitmapPropsAlt && !getObjectFromHitmapIdAlt) || (!getHitmapPropsAlt && getObjectFromHitmapIdAlt)) {
+      console.error(
+        "Possible wrong hitmap id mapping in the instanced rendering. Keep or remove both `getHitmapProps` and `getObjectFromHitmapId` props."
+      );
+    }
+
+    if (enableInstanceHitmap && (!getHitmapProps || !getObjectFromHitmapId)) {
+      getHitmapProps = null;
+      getObjectFromHitmapId = null;
+      console.error(`Default instanced hitmap for ${name} is not supported yet.`);
+    }
+
+    // TODO: deprecating, remove before 1.x release
+    if (getHitmapId) {
+      console.warn(
+        `"getHitmapId" is deprecated. Check "${name}" to use "enableHitmap" to turn on the default hitmap mapping or set the "getHitmapProps" and "getObjectFromHitmapId" props explicitly. `
+      );
+      hitmapProps = defaultGetHitmapProps(getHitmapId, children);
+      getObjectFromHitmapId = defaultGetObjectFromHitmapId;
+    } else if (getHitmapProps && getObjectFromHitmapId) {
+      if (enableHitmap) {
+        hitmapProps = getHitmapProps(children, false);
+      } else if (enableInstanceHitmap) {
+        hitmapProps = getHitmapProps(children, true);
+      }
+    }
```

Comment:
this method feels realllly complicated. is it because we want to preserve compatibility with pre-existing APIs?

I wonder if this can be handled elsewhere, since it's really difficult to read (it also seems brittle and would require testing).